app/controllers/users_controller.rb

```diff
@@ -0,0 +1,17 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @users = User.find(params[:id])
+  end
+
+  def show_post
+    @posts = Post.find(params[:id])
+  end
+
+  def post
+    @posts = Post.all
+  end
```

Comment:
- [ ] Please make sure to create a posts controller separate and place these actions inside the posts controller.